### PR TITLE
Upcoming and current contract agreement extension

### PR DESCRIPTION
**Pull request by tarikstafford**

### CODE REVIEW

File: src/resolvers/cross-schema/quoteDetailsTable.ts [hunk 1]
```diff
@@ -42,61 +43,9 @@ export const crossSchemaExtensions = `
...
-            }
-            ... on DanishAccidentDetails {
-                coInsured
-            }
-            ... on DanishTravelDetails {
-                coInsured
-            }
-          }
-        }`,
+        fragment: fragment,
```
Comment (sampettersson) [inline src/resolvers/cross-schema/quoteDetailsTable.ts:48]:
```suggestion
        fragment,
```

File: src/resolvers/cross-schema/index.ts [hunk 2]
```diff
@@ -488,8 +488,8 @@ export const getCrossSchemaResolvers = (
           })
         },
       },
-      ...quoteDetailsTable("BundledQuote"),
-      ...quoteDisplayName("BundledQuote")
+      ...quoteDetailsTable(getQuoteDetailsFragment("BundledQuote")),
+      ...quoteDisplayName(getQuoteDetailsFragment("BundledQuote"))
```
Comment (sampettersson) [inline src/resolvers/cross-schema/index.ts:492]:
```suggestion
      ...quoteDisplayName("BundledQuote")
```

File: src/resolvers/cross-schema/index.ts [hunk 3]
```diff
@@ -405,8 +405,8 @@ export const getCrossSchemaResolvers = (
           })
         },
       },
-      ...quoteDetailsTable("CompleteQuote"),
-      ...quoteDisplayName("CompleteQuote")
+      ...quoteDetailsTable(getQuoteDetailsFragment("CompleteQuote")),
+      ...quoteDisplayName(getQuoteDetailsFragment("CompleteQuote"))
```
Comment (sampettersson) [inline src/resolvers/cross-schema/index.ts:409]:
```suggestion
      ...quoteDisplayName("CompleteQuote")
```

File: src/resolvers/cross-schema/index.ts [hunk 4]
```diff
@@ -533,3 +533,102 @@ export const getCrossSchemaResolvers = (
...
+//     }
+//     ... on DanishTravelAgreement {
+//         street
+//         zipCode
+//         coInsured
+//     }
+//   }
+// }`
+
+function getQuoteDetailsFragment(quoteType:String): String {
```
Comment (sampettersson) [inline src/resolvers/cross-schema/index.ts:580]:
```suggestion
const getQuoteDetailsFragment = (quoteType: String) => {
```

**Review by sampettersson — CHANGES_REQUESTED** [review]

File: src/resolvers/cross-schema/index.ts [hunk 5]
```diff
@@ -324,6 +324,172 @@ export const getCrossSchemaResolvers = (
           })
         },
       },
```
Comment (hugokallstrom) [inline src/resolvers/cross-schema/index.ts:326]: Do we want these fragment definitions in `quoteDetailsTable` maybe? As `const`, `enum` or something else?

File: src/resolvers/cross-schema/quoteDetailsTable.ts [hunk 6]
```diff
@@ -41,6 +40,10 @@ export const crossSchemaExtensions = `
     extend type BundledQuote {
         detailsTable(locale: Locale!): Table!
     }
+    
+    extend type Contract {
+      detailsTable(locale: Locale!): Table!
```
Comment (sampettersson) [inline src/resolvers/cross-schema/quoteDetailsTable.ts:45]: should we maybe have a `currentAgreementDetailsTable` and a `upcomingAgreementDetailsTable`? where current looks at current agreement and upcoming looks at upcomingAgreement, then its also easy to diff client-side to only show the changes.

Comment (hugokallstrom) [reply src/resolvers/cross-schema/quoteDetailsTable.ts:45]: Ok so rename `Contract` extension to `currentAgreementDetailsTable`, and `CompleteQuote` + `BundledQuote` to `upcomingAgreementDetailsTable`. Should we have those two as separate `crossSchemaExtensions` and re-use the resolve function?

Comment (sampettersson) [reply src/resolvers/cross-schema/quoteDetailsTable.ts:45]: @hugokallstrom no, not really, quotes are fine!

On contracts we have both the `Contract.currentAgreement` that we use on the contract subview to show the details + this new `ContractStatus.upcomingAgreement` that is used after the user has accepted the new quotes right?

Comment (hugokallstrom) [reply src/resolvers/cross-schema/quoteDetailsTable.ts:45]: Ah yes thats right, I'll add that!

**Review by sampettersson — APPROVED** [review]

File: src/resolvers/cross-schema/detailstable/tableRenderer.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import { TableRow } from './table'
+
+export class TableRenderer {
+
+    textKeyMap: any
+    private labelMap: Record<string, string>
+    private valueRenderer: Record<string, (value: any) => string>
+
+    constructor(locale: String) {
+        this.textKeyMap = require(`${__dirname}/../../../translations/${locale}.json`)
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply src/resolvers/cross-schema/detailstable/tableRenderer.ts:10]: Yes, it seems to work :)

**Pull request merged**